gordo_components/data_provider/iroc_reader.py

```diff
@@ -53,7 +55,21 @@ def load_series(
                 f"Iroc reader called with to_ts: {to_ts} before from_ts: {from_ts}"
             )
 
-        base_path = base_path.strip("/")
+        base_paths_from_assets = list(
+            map(lambda tag: IrocReader.base_path_from_asset(tag.asset), tag_list)
+        )
+        if len(set(base_paths_from_assets)) > 1:
+            logger.warning(
+                "Iroc reader found more than one asset from the tag list provided, returning none."
+            )
+            return
+        elif len(base_paths_from_assets) != len(tag_list):
+            logger.warning(
+                "Iroc reader could not associate some tags to an asset, returning none."
+            )
+            return
+
+        base_path = base_paths_from_assets[0]
```

Comment:
You also need to handle the "None" case - where the resolved paths are "None". That would go through as it is now.